### DISCUSSION TITLE
Any way to track neovim release compatibility and pin nvim-treesitter before it gets ahead of the release?

### DISCUSSION THREAD

**Discussion by Lattay** [nvim-treesitter/nvim-treesitter#6896] [General]
**Is your feature request related to a problem? Please describe.**
When using an official release of neovim rather than nightly, it may happen that nvim-treesitter gets ahead of the state of the treesitter support in neovim causing bugs and repeated errors.

**Describe the solution you'd like**
It would be nice to have a way to track which commit of nvim-treesitter is compatible with the last nvim release.
For example, as far as I can tell, any commit after eab58e9b causes errors when editing latex files in neovim 0.10.0 (patch 5 from archlinux repos).
Pining nvim-treesitter to this commit is fine with me, but it requires some research to identify it.

**Describe alternatives you've considered**
I could use nightly, but then other plugins are buggy, and I'd rather use the stable release provided by my distro.

**EDIT:** Sorry I missed the fact that nvim-treesitter seems to be tagged following neovim releases because there is no 0.10.0 tag yet. Maybe tag eab58e9b as 0.10.0?

**Comment by clason** [comment]
This has nothing to do with Neovim releases (the minimum version is clearly documented in the README, and we do tag the last compatible commit before raising the version). Any _parser_ incompatibility is due to a downstream plugin not updating their queries, and the issue has to be reported there.

(It could also be that arch ships their own treesitter parsers and puts them in runtimepath _before_ your nvim-treesitter directory; in this case either your directory is in a non-standard path -- probably due to lazy-loading or distribution shenanigans-- or arch messes with runtimepath and you should complain to them loudly until they stop.)

**Reply by clason** [reply]
> Sorry I missed the fact that nvim-treesitter seems to be tagged following neovim releases because there is no 0.10.0 tag yet. Maybe tag https://github.com/nvim-treesitter/nvim-treesitter/commit/eab58e9bff2a97448ad05b6bd025f6829bb07095 as 0.10.0?

We do not make releases (yet), just tags, whose number is **NOT** related to any Neovim release; it's just a simple incrementing counter.

**Reply by Lattay** [reply]
> the minimum version is clearly documented in the README, and we do tag the last compatible commit before raising the version

Sorry, I have seen that, but since I have errors related to treesitter in 0.10.0 it seemed outdated to me.

> Any _parser_ incompatibility is due to a downstream plugin not updating their queries, and the issue has to be reported there.

I don't have other plugins providing queries. Besides the errors where identified by neovim in the query files shipped with nvim-treesitter and checking out eab58e9b solved my problems.

> It could also be that arch ships their own treesitter parsers

They do, but they do not ship the latex parser, with which I had problems.

> It could also be that arch ships their own treesitter parsers and puts them in runtimepath before your nvim-treesitter directory; in this case either your directory is in a non-standard path -- probably due to lazy-loading or distribution shenanigans-- or arch messes with runtimepath and you should complain to them loudly until they stop.

Parsers installed with TSInstall are prioritized over the ones distributed by arch.
Proof is that the vimdoc parser is installed by arch and is incompatible with nvim-treesitter in eab58e9b but doing a `:TSInstall vimdoc` solves the errors.

Actually, I think I may have found my problem. **In the absence of nodejs on my machine the latex parser stopped being updated somehow, but the previous version was still available.** Removing nvim-treesitter and reinstalling it pointed to that problem when the plugin tried to recompile the latex parser.

Sorry for the noise.
